egs/csj/ASR/local/utils/tokenizer.py

```diff
@@ -0,0 +1,247 @@
+import argparse
+from pathlib import Path
+from typing import Callable, List, Union
+
+import sentencepiece as spm
+from k2 import SymbolTable
+
+
+class Tokenizer:
+    text2word: Callable[[str], List[str]]
+
+    @staticmethod
+    def add_arguments(parser: argparse.ArgumentParser):
+        group = parser.add_argument_group(title="Lang related options")
+
+        group.add_argument("--lang", type=Path, help="Path to lang directory.")
+
+        group.add_argument(
+            "--lang-type",
+            type=str,
+            default=None,
+            help="Either 'bpe', 'char', or 'phone'. If not provided, it expects lang_dir/lang_type to exists.",
```

Comment:
```suggestion
            help="Either 'bpe' or 'char'. If not provided, it expects lang_dir/lang_type to exists.",
```

Looks like `phone` is not supported at present.

Reply:
Thanks for the reminder. Currently this recipe does not support BPE too. I will make those changes.